lab_2/main.py

```diff
@@ -34,7 +69,23 @@ def fill_lcs_matrix(first_sentence_tokens: tuple, second_sentence_tokens: tuple)
     :param second_sentence_tokens: a tuple of tokens
     :return: a lcs matrix
     """
-    pass
+    incorrect_inputs = is_inputs_incorrect(first_sentence_tokens, second_sentence_tokens)
+    if incorrect_inputs or not first_sentence_tokens or not second_sentence_tokens:
+        return []
+
+    lcs_matrix = create_zero_matrix(len(first_sentence_tokens) + 1, len(second_sentence_tokens) + 1)
+    for first_i, first_elem in enumerate(first_sentence_tokens):
+        for sec_i, second_elem in enumerate(second_sentence_tokens):
+            if first_elem == second_elem:
+                lcs_matrix[first_i][sec_i] = lcs_matrix[first_i - 1][sec_i - 1] + 1
+            else:
+                lcs_matrix[first_i][sec_i] = max(lcs_matrix[first_i][sec_i - 1], lcs_matrix[first_i - 1][sec_i])
+
+    del lcs_matrix[-1]
+    for row in lcs_matrix:
+        del row[-1]
```

Comment:
Please, provide some explanation.

Reply:
I needed to delete zero-row and -column to pass the tests, but now I don't need them anymore